vllm/v1/spec_decode/eagle.py

```diff
@@ -347,6 +353,18 @@ def dummy_run(
                 hidden_states=self.hidden_states[:num_tokens],
             )
 
+    def validate_kv_cache_group(self, kv_cache_config: KVCacheConfig) -> None:
+        kv_cache_groups: dict[str, int] = {}
+        for id, kv_cache_group in enumerate(kv_cache_config.kv_cache_groups):
+            for layer_name in kv_cache_group.layer_names:
+                kv_cache_groups[layer_name] = id
+        assert len(
+            set([
+                kv_cache_groups[layer_name]
+                for layer_name in self.attn_layer_names
+            ])) == 1, "For multi-layer eagle draft model, "
+        "all layers should belong to the same kv cache group"
```

Comment:
```suggestion
            ])) == 1, "All eagle layers should belong to the same kv cache group"
```
I think we don't need to specify multi-layer here. This constraint also hold for single layer case.